File: busqueda lineal 01-09-25/lineal.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function busquedaLineal(arreglo, objetivo) {
    for (let i = 0; i < arreglo.length; i++) {
        if (arreglo[i] === objetivo) {
            return i; // Retorna el índice si se encuentra el elemento
        }
    }
    return -1; // Retorna -1 si el elemento no se encuentra
}

function imprimirArreglo(arreglo) {
    // Usamos .join() para un formato más limpio en JS
    console.log("Arreglo: " + arreglo.join(' '));
}

function insertarNumero(arreglo, numero, posicion) {
    // Validamos que la posición esté dentro de los límites del arreglo
    if (posicion < 0 || posicion >= arreglo.length) {
        console.log("Posición inválida.");
        return;
    }
    arreglo[posicion] = numero;
}

const arreglo = [10, 20, 30, 40, 50];
imprimirArreglo(arreglo);

// Creamos una función asíncrona para poder usar await
async function main() {
    try {
        const objetivoStr = await new Promise(resolve => rl.question('Introduce el elemento a buscar: ', resolve));
        const objetivo = parseInt(objetivoStr);

        // Validamos que la entrada sea un número
        if (isNaN(objetivo)) {
            console.log("Error: Debes introducir un número válido.");
            return;
        }

        const indice = busquedaLineal(arreglo, objetivo);
        if (indice !== -1) {
            console.log(`Elemento encontrado en el índice: ${indice}`);
        } else {
            console.log("Elemento no encontrado.");
        }

        const nuevoValorStr = await new Promise(resolve => rl.question('Valor para insertar: ', resolve));
        const nuevoValor = parseInt(nuevoValorStr);

        const posicionStr = await new Promise(resolve => rl.question('Posición para insertar: ', resolve));
        const posicion = parseInt(posicionStr);

        // Validamos que las entradas para insertar sean números
        if (isNaN(nuevoValor) || isNaN(posicion)) {
            console.log("Error: El valor y la posición deben ser números válidos.");
            return;
        }

        insertarNumero(arreglo, nuevoValor, posicion);
        imprimirArreglo(arreglo);

    } catch (error) {
        console.error("Ocurrió un error:", error);
    } finally {
        // Nos aseguramos de cerrar readline al final
        rl.close();
    }
}

// Llamamos a nuestra función principal para que se ejecute
main();

File: busqueda lineal 01-09-25/lineal.cpp
#include <iostream>

using namespace std;
void imprimirArreglo(int arr[], int size) {
    cout << "Contenido del arreglo: ";
    for (int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}
int busquedaLineal(int arr[], int size, int objetivo) {
    for (int i = 0; i < size; i++) {
        if (arr[i] == objetivo) {
            return i; // Retorna el índice si se encuentra el elemento
        }
    }
    return -1; // Retorna -1 si no se encuentra el elemento
}
void insertarnumero(int arr[], int nuevoValor, int posicion, int size) {

    if (posicion < 0 || posicion > size) {
        cout << "Posición inválida." << endl;
        return;
    }
    for (int i = size - 1; i > posicion; i--) {
        arr[i] = arr[i - 1];
    }
    arr[posicion] = nuevoValor;
}

int main() {
    int arreglo[100] = {10, 20, 30, 40, 50};
    int size = 5;
    imprimirArreglo(arreglo, size);

    int objetivo;
    cout << "Introduce el elemento a buscar: ";
    cin >> objetivo;

    int indice = busquedaLineal(arreglo, size, objetivo);
    if (indice != -1) {
        cout << "Elemento encontrado en el índice: " << indice << endl;
    } else {
        cout << "Elemento no encontrado." << endl;
    }
    int nuevoValor, posicion;
    cout << "Valor para insertar: ";
    cin >> nuevoValor;
    cout << "Posición para insertar: ";
    cin >> posicion;
    insertarnumero(arreglo, nuevoValor, posicion, size);
    imprimirArreglo(arreglo, size);
    cout << "Fin del programa." << endl;

    return 0;
}